pc-bufsw.el

```diff
@@ -256,10 +263,16 @@ Buffers are odered from most to least recently used.")
 
 (defun pc-bufsw--get-walk-vector ()
   ;; Construct main buffer vector.
-  (let* ((cur-buf (current-buffer))
-	 (assembled (list cur-buf)))
-    (dolist (buf (buffer-list))
-      (when (and (not (eq buf cur-buf))
+  (let* (assembled
+	 (buffers (append
+		   (list (current-buffer))
+		   (when pc-bufsw-prefer-current-window
+		     (mapcar #'car (window-prev-buffers)))
+		   (when pc-bufsw-prefer-current-frame
+		     (frame-parameter (selected-frame) 'buffer-list))
+		   (buffer-list))))
+    (dolist (buf buffers)
```

Comment:
This is O(N^2) algorithm. If one have hundreds of buffers, it can be an issue. Use a hash-table to recorded already visited buffers.

Reply:
Hash tables only begin to be faster after 1500 buffers according to this benchmark:

```elisp
(eval-when-compile
  (require 'cl))

(cl-loop for num-buffers from 0 to 10000 by 100
         do
         (let (buffers)
           (message "Creating %d buffers..." num-buffers)
           (setq buffers
                 (mapcar (lambda (n)
                           (get-buffer-create (format " test-%d" n)))
                         (number-sequence 1 num-buffers)))
           (message
            "List: %s"
            (benchmark-run 100
              (let (seen)
                (mapc (lambda (b)
                        (unless (memq b seen)
                          (setq seen (cons b seen))))
                      buffers))))
           (message
            "Hash: %s"
            (benchmark-run 100
              (let ((hash (make-hash-table)))
                (mapc (lambda (b)
                        (unless (gethash b hash)
                          (puthash b t hash)))
                      buffers))))))
```

Garbage collection seems to be a large culprit.

Even at 10000 buffers, hash tables are only 3 times faster.

Is it worth it?

Reply:
Hashes do get faster when preallocated with the size as the buffer count, but only significantly so in the thousands of buffers range.

Reply:
For the sanity then lets restrict the number of buffers to show to 100 and keep the nested loops.